azurelinuxagent/common/exception.py

```diff
@@ -65,6 +65,8 @@ class ExtensionError(AgentError):
     def __init__(self, msg=None, inner=None, code=-1):
         super(ExtensionError, self).__init__(msg, inner)
         self.code = code
+        # Setting log_event to always log by default
+        self.log_event = True
```

Comment:
since now the exception always has the log_event attribute, this check on handle_ext_handler_error is not needed:
>     log_event = e.log_event if hasattr(e, "log_event") else True